### PR TITLE
docs - instructions for running development branch

### PR DESCRIPTION
**Pull request by frannietrempe**
- This was a Notion doc, revised and reorganized it a bit to add to developer's guide in docs.
- Added mentions on developer's guide start page and install guide doc.

### CODE REVIEW

**Comment by ariya** [comment]
Any particular reason why the contents in `running-dev-branch.md` not being merged into `devenv.md`? There's a lot of similarity (after all, it's about developing Metabase).

Some of the (detailed) info regarding the prerequisite (Xcode on Mac etc) can be spawned into its own `.md` file, since then it can be referred from `build.md`.

File: docs/developers-guide/build.md [hunk 1]
```diff
@@ -1,25 +1,132 @@
...
+
+If you have multiple JDK versions installed in your machine, be sure to switch your JDK before building with: 
+
+```
+sudo update-alternatives --config java
+``` 
+
+Then select Java 11 in the menu.
 
 If you are developing on Windows, make sure to use Ubuntu on Windows and follow instructions for Ubuntu/Linux instead of installing ordinary Windows versions.
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:41]:
```suggestion
### If you're running Windows, use WSL

If you are developing on Windows, you should run Ubuntu on Windows Subsystem for Linux (WSL) and follow instructions for Ubuntu/Linux.
```

File: docs/developers-guide/build.md [hunk 2]
```diff
@@ -1,25 +1,132 @@
...
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:42]:
```suggestion
### Developing with VS Code in a remote container
```

File: docs/developers-guide/build.md [hunk 3]
```diff
@@ -1,25 +1,132 @@
...
 
 Alternatively, without the need to explicitly install the above dependencies, follow the guide [on using Visual Studio Code](deven.md#developing-with-visual-studio-code.md) and its remote container support.
 
+## Copy the Metabase code to your computer
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:45]:
```suggestion
## Clone the Metabase repo
```

File: docs/developers-guide/build.md [hunk 4]
```diff
@@ -1,25 +1,132 @@
...
+
+Once you have all that installed, you’re going to get a copy of Metabase's source code from GitHub, where it’s stored.
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:47]:
```suggestion
Once you've installed all the build tools, you'll need to clone the Metabase repository from GitHub.
```

File: docs/developers-guide/build.md [hunk 5]
```diff
@@ -1,25 +1,132 @@
...
+
+1. Create a `workspace` folder (you can name it that or whatever you want). You’ll put the Metabase code files inside this `workspace` folder.
+
+2. Open up your terminal app, and navigate to your workspace folder with: 
+
+```
+cd ~/workspace
+```
+
+1. Run this GitHub command to “clone” Metabase into this folder
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:57]: Do they need to set up the `gh` CLI to do this? Should that be in the required tools?

https://cli.github.com/manual/

And should we provide the alternate git clone command?

File: docs/developers-guide/build.md [hunk 6]
```diff
@@ -1,25 +1,132 @@
...
+   cd ~/workspace/metabase
+   ```
+
+3. "Pull” down the latest code by running: 
+
+   ```
+   git pull
+   ```
+
+   You should do this every time to make sure that you have all the latest Metabase branches and code on your computer. It’s also how you’ll get updates on a feature branch if the engineer make changes to it.
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:83]:
```suggestion
   You should do this every time to make sure that you have all the latest Metabase branches and code on your computer. It’s also how you’ll get updates on a feature branch someone make changes to it.
```

File: docs/developers-guide/build.md [hunk 7]
```diff
@@ -1,25 +1,132 @@
...
+
+4. Find the name of the branch you want to run by going to the “pull request” page for that feature on GitHub and copying the branch name from there. Here’s [an example PR page](https://github.com/metabase/metabase/pull/19138), with the branch name
+`fix-native-dataset-drill-popover`.
+
+5. Switch to, or “check out,” that branch by running:
+
+   ```
+   git checkout branch-name
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:91]:
```suggestion
   git checkout <branch-name>
```

File: docs/developers-guide/build.md [hunk 8]
```diff
@@ -1,25 +1,132 @@
...
+   git checkout fix-native-dataset-drill-popover
+   ```
+    
+   When you want to switch back to `master`, run: 
+    
+   ```
+   git checkout master
+   ```
+
+6. Now we’ll start up the backend server of Metabase with:
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:106]: This should be its own section, as it talks about running the code (not getting the code).

File: docs/developers-guide/build.md [hunk 9]
```diff
@@ -1,25 +1,132 @@
...
+
+A couple useful tips:
+
+- To switch to a different branch or back to `master`, open up another Terminal tab, and repeat steps 3, 4, and 5. You’ll need to do steps 6 and 7 too if Metabase wasn’t already running. If it was already running, the frontend will automatically rebuild itself. You can check its progress by switching to that tab in your Terminal — it usually takes something like 15 seconds, but will depend on your hardware.
+
+- If you want to make Metabase stop running, you can either quit your terminal program, or go to the tab with the backend running and hit `Ctrl+C` to stop the backend. Most of the time you don’t have to do this to switch branches, but there are some cases where the change or feature you’re trying to see is a change with the backend, and you may need to stop the backend with `Ctrl+C` and then restart it by completing Step 6 again.
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:128]:
```suggestion

## Shutting down Metabase

If you want to make Metabase stop running, you can either quit your terminal program, or go to the tab with the backend running and hit `Ctrl+C` to stop the backend. Most of the time you don’t have to do this to switch branches, but there are some cases where the change or feature you’re trying to see is a change with the backend, and you may need to stop the backend with `Ctrl+C` and then restart it by completing Step 6 again.
```

File: docs/developers-guide/build.md [hunk 10]
```diff
@@ -1,25 +1,132 @@
...
+
 ## Build Metabase Uberjar
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:130]:
```suggestion
## Building the Metabase Uberjar
```

File: docs/developers-guide/dev-branch-docker.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+OR
+
+- If you like to install things via Homebrew:
+
+```bash
+brew install docker
+```
+Once Docker is installed, you’re ready to go.
+
+## Run an in development branch to test or verify features
```
Comment (jeff-bruemmer) [inline docs/developers-guide/dev-branch-docker.md:22]:
```suggestion
## Run a development branch to test or verify features
```

**Review by jeff-bruemmer — CHANGES_REQUESTED** [review]
Just some suggestions.

**Review by jeff-bruemmer — APPROVED** [review]
Thanks, @frannietrempe!

**Comment by frannietrempe** [comment]
a belated thank you to @ariya for pointing out that this info could be integrated into existing pages 🙏

**Pull request merged**